src/main/java/uk/gov/companieshouse/api/accounts/controller/CurrentPeriodController.java

```diff
@@ -26,9 +26,10 @@
     public ResponseEntity create(@Valid @RequestBody CurrentPeriod currentPeriod,
             HttpServletRequest request)
             throws NoSuchAlgorithmException {
-        Transaction transaction = (Transaction) request.getSession().getAttribute(AttributeName.TRANSACTION.getValue());
+        CompanyAccountEntity companyAccountEntity = (CompanyAccountEntity) request.getSession()
+                .getAttribute(AttributeName.COMPANY_ACCOUNT.getValue());
         CurrentPeriod result = currentPeriodService
-                .save(currentPeriod, transaction.getCompanyNumber());
+                .save(currentPeriod, companyAccountEntity.getId());
```

Comment:
similar to https://github.com/companieshouse/company-accounts.api.ch.gov.uk/pull/18/files we should be checking for error conditions here while writing to the database.

This PR has been merged into develop so the 'ResponseObject' pattern seen there should be used when calling a service from a controller.